### PR TITLE
Link api docs in examples for Mesh Creation

### PR DESCRIPTION
**Pull request by MatthewFlamm**
### Overview

Start for #5111

### Details

I went through "00-load" folder, AKA "Mesh Creation", and tried to add some links to pyvista functions/classes that are primarily showcased in each.

### CODE REVIEW

**Comment by MatthewFlamm** [comment]
@pyvista-bot preview

File: examples/00-load/create-geometric-objects.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 Geometric Objects
 ~~~~~~~~~~~~~~~~~
 
-The "Hello, world!" of VTK
+The "Hello, world!" of VTK.  Uses objects from :ref:`geometry_api`.
```
Comment (tkoyama010) [inline examples/00-load/create-geometric-objects.py:7]:
```suggestion
The "Hello, world!" of VTK.
Uses objects from :ref:`geometry_api`.
```

File: examples/00-load/create-polyhedron.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 This example shows how to build a simple :class:`pyvista.UnstructuredGrid`
 using polyhedra. We will be using VTK types to determine which type of cells we
-are building.
+are building.  A list of cell types is given in :class:`pyvista.CellType`.
```
Comment (tkoyama010) [inline examples/00-load/create-polyhedron.py:9]:
```suggestion
are building. 
A list of cell types is given in :class:`pyvista.CellType`.
```

File: examples/00-load/create-structured-surface.py [hunk 3]
```diff
@@ -4,7 +4,8 @@
 Creating a Structured Surface
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Create a StructuredGrid surface from NumPy arrays
+Create a StructuredGrid surface from NumPy arrays using
+:class:`pyvista.StructuredGrid`.
```
Comment (tkoyama010) [inline examples/00-load/create-structured-surface.py:8]:
```suggestion
Create a StructuredGrid surface from NumPy arrays
using :class:`pyvista.StructuredGrid`.
```

File: examples/00-load/create-polyhedron.py [hunk 4]
```diff
@@ -6,7 +6,8 @@
 
 This example shows how to build a simple :class:`pyvista.UnstructuredGrid`
 using polyhedra. We will be using VTK types to determine which type of cells we
-are building.
+are building. 
+A list of cell types is given in :class:`pyvista.CellType`.
```
Comment (tkoyama010) [inline examples/00-load/create-polyhedron.py:10]:
```suggestion
are building. A list of cell types is given in :class:`pyvista.CellType`.
```

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: examples/00-load/create-uniform-grid.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 Create a simple uniform grid from a 3D NumPy array of values.
+This example uses :class:`pv.ImageData`.
```
Comment (MatthewFlamm) [inline examples/00-load/create-uniform-grid.py:6]:
```suggestion
This example uses :class:`pyvista.ImageData`.
```

**Review by tkoyama010 — APPROVED** [review]

File: examples/00-load/create-kochanek-spline.py [hunk 6]
```diff
@@ -5,6 +5,7 @@
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
 Create a Kochanek spline/polyline from a numpy array of XYZ vertices.
+Uses :func:`pyvista.KochanekSpline`.
```
Comment (MatthewFlamm) [inline examples/00-load/create-kochanek-spline.py:8]: This doesn't link, I think it is missing from the docs.  Not for this PR

Comment (MatthewFlamm) [reply examples/00-load/create-kochanek-spline.py:8]: Decided to add it

**Comment by tkoyama010** [comment]
@pyvista-bot preview

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**